### DISCUSSION TITLE
Allow dump and die tag outside of Twig blocks when extending a template

### DISCUSSION THREAD

**Discussion by simonkuran** [craftcms/cms#10867] [Ideas] [answered]
When extending a layout in Twig it would be a great developer experience feature if {% dd someVariable %} could be used anywhere without getting an error.

For example:
```
{% extends '_layouts/defaultLayout' %}

{# Code to generate someVariable #}

{% dd someVariable %}

{# Lots more code #}

{% block pageContent %}

{% endblock %}
```
Returns this error:
"_A template that extends another one cannot include content outside Twig blocks. Did you forget to put the content inside a {% block %} tag?"_

The error message makes sense, but I wonder if there's a way to make an exception for dump and die since's it's highjacking how the page renders anyway. Sometimes I have 100 lines of code before get to the template's block tag and it would be handy to be able to quickly debug things anywhere in the file.

**Comment by brandonkelly** [comment, marked as answer]
I’m not able to reproduce this – `{% dd %}` tags are working fine even within child templates. Are you sure it’s the `{% dd %}` tag that’s causing the error for you?

**Reply by simonkuran** [reply]
I think I have it right, it's super basic and this is with Craft v3.7.37.

Outputs 'test':
```
{% extends '_layouts/defaultLayout' %}
{% block pageContent %}
<pre>{% dd 'test' %}</pre>
{% endblock %}
```

Outputs the error message:
```
{% extends '_layouts/defaultLayout' %}
<pre>{% dd 'test' %}</pre>
{% block pageContent %}
{% endblock %}
```

For both of these tests _layouts/defaultLayout is only one line:
```
{{ block('pageContent') }}
```

**Reply by brandonkelly** [reply]
The problem is your `<pre>` tag. That’s what’s causing the error, and it’s not necessary – `{% dd %}` will output its own `<pre>` tag.

```twig
{% extends '_layouts/defaultLayout' %}
{% dd 'test' %}
{% block pageContent %}
{% endblock %}
```

**Reply by simonkuran** [reply]
Ah, makes sense. Thanks for the help!
